### ISSUE TITLE
Implement minimal subset of `any[]`

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/nballerina#135]
This does arrays of `any` without mutation.

Includes:

* New type descriptors: `any[]`
* New expressions
   * list constructors `[1, 2, 3]` - syntactically allowed only where there is an explicit contextually expected type e.g. as function argument, initializer or following a type cast
   * `E[n]`

Existing syntax extended:
* `===`, `!==`
* `io:println`

Note: at this point `io:println` does not have to deal with cycles.

Depends on #131 

Adding `array:length` will be done in #131.

**Comment by jclark** [comment]
We can split this into the following steps:

- [x] construct a list and print it out with io:println (I think code for identity will work as is)
- [x] add `any[]` type descriptor
- [x] make `E[n]` work
- [x] check that `===` works properly
- [x] check that `==` is not allowed

**Comment by jclark** [comment]
See https://github.com/ballerina-platform/nballerina/issues/157#issuecomment-871944555 for bounds checking on `E[n]`: we can do a single unsigned compare of the i64 index against the length.
